src/fsharp/NicePrint.fs

```diff
@@ -1505,6 +1505,7 @@ module private TastDefinitionPrinting =
                                       not (isInterfaceTy denv.g oty)
                                   | [] -> true)
               |> List.filter (fun v -> denv.showObsoleteMembers || not (HasFSharpAttribute denv.g denv.g.attrib_SystemObsolete v.Attribs))
+              |> List.filter (fun v -> not (Infos.AttributeChecking.CheckFSharpAttributesForHidden denv.g v.Attribs))
```

Comment:
The NicePrint.fs code is used to show the signature of things when emitting signature files using the "-sig:file" option. So I think you need to show obsolete, hidden and experimental members in that case, because the signature file should be "logically complete".  

For this, you should add a flag "showHiddenMembers" to DisplayEnv with default value "false", and set this to "true" in BuildInitialDisplayEnvForDocGeneration.

I actually think "showObsoleteMembers" should match this and be "false" by default, but set to "true" in BuildInitialDisplayEnvForDocGeneration.  I don't think we want obsolete members shown in class/module tooltips.

Finally, rename BuildInitialDisplayEnvForDocGeneration to BuildInitialDisplayEnvForSigFileGeneration.